torch/csrc/distributed/c10d/reducer.cpp

```diff
@@ -2374,8 +2374,10 @@ void Reducer::reset_state() {
 
   // Reset unused parameter accounting.
   // See Note [local_used_map_ -> local_used_map_dev copying]
-  local_used_map_.fill_(0);
-  local_used_map_reduced_ = false;
+  if (find_unused_parameters_) {
+    local_used_map_.fill_(0);
```

Comment:
nit: I do not write C++ PyTorch, but could this just be `local_used_map_.zero_()`?